### PR TITLE
Feature/gazebo9 voliro px4

### PR DESCRIPTION
**Pull request by kajabo**
Mavlink interface plugin in rotors_gazebo_plugins now considers flags, and interprets them as commands for the overactuated hexacopter 'voliro'.

- Max number of actuators now 18 instead of 16
- Flag 1 for rotor speeds 1 to 12
- Flag 2 for tilt angles 1 to 6
- Other flag values default to previous behaviour

### CODE REVIEW

**Comment by michaelpantic** [comment]
Just checked, built on Gazebo7.0 and 9.0 - so no changes in autobackport script necessary for now.

File: rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_mavlink_interface.h [hunk 1]
```diff
@@ -153,7 +153,7 @@ class GazeboMavlinkInterface : public ModelPlugin {
   void handle_message(mavlink_message_t *msg);
   void pollForMAVLinkMessages(double _dt, uint32_t _timeoutMs);
 
-  static const unsigned kNOutMax = 16;
+  static const unsigned kNOutMax = 18;
```
Comment (ffurrer) [inline rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_mavlink_interface.h]: use `constexpr` (`uint32_t` or `size_t` or whatever)...
```C++
constexpr size_t kNOutMax = 18u;
```

File: rotors_gazebo_plugins/CMakeLists.txt [hunk 2]
```diff
@@ -23,8 +23,8 @@ project(rotors_gazebo_plugins)
 # =============================================================================================== #
 
 if(NOT DEFINED BUILD_MAVLINK_INTERFACE_PLUGIN)
-  message(STATUS "BUILD_MAVLINK_INTERFACE_PLUGIN variable not provided, setting to FALSE.")
-  set(BUILD_MAVLINK_INTERFACE_PLUGIN FALSE)
+  message(STATUS "BUILD_MAVLINK_INTERFACE_PLUGIN variable not provided, but anyway setting to TRUE.")
```
Comment (ffurrer) [inline rotors_gazebo_plugins/CMakeLists.txt]: This doesn't make sense to change here, we can set the variable to TRUE by default but not like this ;) (why do you want it to be TRUE by default?)

File: rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp [hunk 3]
```diff
@@ -952,20 +958,54 @@ void GazeboMavlinkInterface::handle_message(mavlink_message_t *msg)
...
-        // if (joints_[i])
-        //   gzerr << i << " : " << input_index_[i] << " : " << controls.controls[input_index_[i]] << " : " << input_reference_[i] << "\n";
-      } else {
-        input_reference_[i] = zero_position_disarmed_[i];
+
+    // Interpret flag 1 as rotor speeds, controller targets for VoliroX system
+    // (12 commands)
+    if (controls.flags == 1) {
+      input_reference_.resize(kNOutMax);
+      for (int i = 0; i < 12; i++) {
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp]: pre-increment (use size_t or other unsigned type?) constexpr for 12.

File: rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp [hunk 4]
```diff
@@ -952,20 +958,54 @@ void GazeboMavlinkInterface::handle_message(mavlink_message_t *msg)
...
-        input_reference_[i] = (controls.controls[input_index_[i]] + input_offset_[i])
-          * input_scaling_[i] + zero_position_armed_[i];
-        // if (joints_[i])
-        //   gzerr << i << " : " << input_index_[i] << " : " << controls.controls[input_index_[i]] << " : " << input_reference_[i] << "\n";
-      } else {
-        input_reference_[i] = zero_position_disarmed_[i];
+
+    // Interpret flag 1 as rotor speeds, controller targets for VoliroX system
+    // (12 commands)
+    if (controls.flags == 1) {
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp]: define constexpr for 1 then you might even be able to drop the comment above.

File: rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp [hunk 5]
```diff
@@ -952,20 +958,54 @@ void GazeboMavlinkInterface::handle_message(mavlink_message_t *msg)
...
+        } else {
+          input_reference_[i] = zero_position_disarmed_[i];
+        }
+      }
+      received_first_reference_ = true;
+    }
+    // Interpret flag 2 as a servo position message for voliroX system
+    // (indices 12 - 17).
+    else if (controls.flags == 2) {
+      for (int i = 12; i < 18; i++) {
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp]: same as above

File: rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp [hunk 6]
```diff
@@ -952,20 +958,54 @@ void GazeboMavlinkInterface::handle_message(mavlink_message_t *msg)
...
+              (controls.controls[input_index_[i]] + input_offset_[i]) *
+                  input_scaling_[i] +
+              zero_position_armed_[i];
+        } else {
+          input_reference_[i] = zero_position_disarmed_[i];
+        }
+      }
+      received_first_reference_ = true;
+    }
+      break;
```
Comment (ffurrer) [inline rotors_gazebo_plugins/src/gazebo_mavlink_interface.cpp]: fix style

**Comment by michaelpantic** [comment]
@kajabo FYI If you address the comments sometime before end of next week, I'll include that into the large release we're doing

**Comment by kajabo** [comment]
@michaelpantic @ffurrer Comments addressed, please check.

**Pull request merged**